include/mega/logging.h

```diff
@@ -125,11 +125,43 @@ class Logger
 public:
     virtual ~Logger() = default;
     // Note: `time` and `source` are null in performance mode
-    virtual void log(const char *time, int loglevel, const char *source, const char *message) = 0;
+    virtual void log(const char *time, int loglevel, const char *source, const char *message
+#ifdef ENABLE_LOG_PERFORMANCE)
+          , std::vector<const char *> directMessages = std::vector<const char *>()
+#endif
+                     ) = 0;
 };
 
 typedef std::vector<std::ostream *> OutputStreams;
 
+/**
+ * @brief holds a const char * and its size to pass to SimpleLogger, to use the direct logging logic
+ */
+class DirectMessage{
+public:
+    size_t mSize = 0;
+    const char *mConstChar;
+
+    DirectMessage( const char *constChar, bool force = false) //force will set size as max, so as to stay above directMsgThreshold
+    {
+        mConstChar = constChar;
+        mSize = force?SIZE_MAX:strlen(constChar);
+    }
+
+    template<typename T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
```

Comment:
I believe `bool` is also considered integral

Reply:
DirectMessage( const char *constChar, bool force = false) is called when passing a `bool`. Otherwise, that second one. The idea is that `DirectMessage( const char *constChar, bool force = false)` is not called when another `is_integral` type is used